ports/rp2/mpbthciport.c

```diff
@@ -50,46 +53,60 @@ void mp_bluetooth_hci_init(void) {
     events_task_is_scheduled = false;
 }
 
-STATIC void mp_bluetooth_hci_start_polling(void) {
-    events_task_is_scheduled = false;
-    mp_bluetooth_hci_poll_now();
-}
-
 static int64_t mp_bluetooth_hci_timer_callback(alarm_id_t id, void *user_data) {
     poll_timer_id = 0;
-    mp_bluetooth_hci_poll_now();
+    mp_bluetooth_hci_schedule_poll();
     return 0;
 }
 
+// The bt stack uses this for command timeouts, so it shouldn't be used elsewhere
 void mp_bluetooth_hci_poll_in_ms(uint32_t ms) {
+    debug_printf("mp_bluetooth_hci_poll_in_ms %lu\n", ms);
+    if (poll_timer_id > 0) {
+        cancel_alarm(poll_timer_id);
```

Comment:
This looks like a bug fix.  Did you find it necessary to add this cancel?

Reply:
I can't remember and the history from last July (!) doesn't seem very informative. It might have come from a time before interrupts were working properly. I'll try removing it.

Reply:
Seems to work ok without it